packages/type-utils/tests/isTypeReadonly.test.ts

```diff
@@ -308,5 +308,35 @@ describe('isTypeReadonly', () => {
         ])('handles non fully readonly sets and maps', runTests);
       });
     });
+
+    describe('allowlist', () => {
+      const options: ReadonlynessOptions = {
+        allowlist: [
+          {
+            from: 'lib',
+            name: 'RegExp',
+          },
+          {
+            from: 'file',
+            name: 'Foo',
+          },
+        ],
+      };
+
+      function runTestIsReadonly(code: string): void {
+        runTestForAliasDeclaration(code, options, true);
+      }
+
+      describe('is readonly', () => {
+        const runTests = runTestIsReadonly;
+
+        it.each([
+          [
+            'interface Bar {readonly prop: RegExp}; type Test = (arg: Bar) => void;',
+          ],
+          ['interface Foo {prop: string}; type Test = (arg: Foo) => void;'],
+        ])('correctly marks allowlisted types as readonly', runTests);
+      });
+    });
```

Comment:
Can we have a couple basic negative/`false` cases too? Since this is the only `isTypeReadonly` test that includes an `allowlist`.